### PR TITLE
Dsp3/zephyr 3.2

### PR DESCRIPTION
**Pull request by lbonhomme-vossloh**
Add Pin configuration for I2S and SPI_CR1_HDDIR management in `LL_I2S_SetTransferMode.

### CODE REVIEW

File: stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h [hunk 1]
```diff
@@ -3029,6 +3029,12 @@ __STATIC_INLINE uint32_t LL_I2S_GetStandard(SPI_TypeDef *SPIx)
 __STATIC_INLINE void LL_I2S_SetTransferMode(SPI_TypeDef *SPIx, uint32_t Standard)
 {
   MODIFY_REG(SPIx->I2SCFGR, SPI_I2SCFGR_I2SCFG, Standard);
+  
+  if(Standard == LL_I2S_MODE_SLAVE_TX || Standard == LL_I2S_MODE_MASTER_TX)
+  	MODIFY_REG(SPIx->CR1, SPI_CR1_HDDIR, SPI_CR1_HDDIR);
```
Comment (smartin-vossloh) [inline stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h]: @lbonhomme-vossloh I do prefer always having brackets surrouding code-block.

It makes easier adding logs (or whatever else), and avoids unexpected expansion on macro calls.

**Review by smartin-vossloh — COMMENTED** [review]
olgtm

File: stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h (hunk 1 above)
Comment (nvincent-vossloh) [reply stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h]: I strongly agree with @smartin-vossloh, I have seen on multiple occasions wasted times due to missing curly braces where they were not mandatory. Save yourself some time and never omit the curly braces :wink:
While you change this line, you can add a space after `if` keyword.

File: dts/st/h7/stm32h743zitx-pinctrl.dtsi [hunk 2]
```diff
@@ -3354,4 +3371,4 @@
 
 		};
 	};
-};
\ No newline at end of file
+};
```
Comment (nvincent-vossloh) [inline dts/st/h7/stm32h743zitx-pinctrl.dtsi]: Your editor is not properly configured and adds new line at the end of the files when saving.

File: stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h [hunk 3]
```diff
@@ -3029,6 +3029,12 @@ __STATIC_INLINE uint32_t LL_I2S_GetStandard(SPI_TypeDef *SPIx)
...
+
+  if(Standard == LL_I2S_MODE_SLAVE_TX || Standard == LL_I2S_MODE_MASTER_TX)
+  	MODIFY_REG(SPIx->CR1, SPI_CR1_HDDIR, 0);
```
Comment (nvincent-vossloh) [inline stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h]: Same as above, add the curly brace and space after `if`.

**Review by nvincent-vossloh — CHANGES_REQUESTED** [review]

File: stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h [hunk 4]
```diff
@@ -3029,6 +3029,13 @@ __STATIC_INLINE uint32_t LL_I2S_GetStandard(SPI_TypeDef *SPIx)
 __STATIC_INLINE void LL_I2S_SetTransferMode(SPI_TypeDef *SPIx, uint32_t Standard)
 {
   MODIFY_REG(SPIx->I2SCFGR, SPI_I2SCFGR_I2SCFG, Standard);
+
+  if(Standard == LL_I2S_MODE_SLAVE_TX || Standard == LL_I2S_MODE_MASTER_TX) {
```
Comment (smartin-vossloh) [inline stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_spi.h]: @lbonhomme-vossloh space after keyword before opening bracket.

Ditto below

**Review by smartin-vossloh — COMMENTED** [review]
olgtm

**Review by smartin-vossloh — COMMENTED** [review]
lgtm

**Pull request merged**